src/ray/util/filesystem.h

```diff
@@ -30,6 +34,21 @@ static char GetPathSep() {
   return result;
 }
 
+/// Returns the executable binary suffix for the platform, if any.
+std::string GetExeSuffix();
+
+/// Equivalent to Python's os.path.basename() for file system paths.
+std::string GetFileName(const std::string &path);
+
+/// Equivalent to Python's os.path.dirname() for file system paths.
+std::string GetParentPath(const std::string &path);
```

Comment:
It looks like this function is currently unused in this PR, how about we remove it?

Reply:
Oh yeah, and same with `GetRootPath` and `TrimDirSep`. Do you think it might become more of a hassle later when people need these but don't find them though? I feel like they might be valuable to keep, because it'll be very easy to write a buggy cross-platform implementation if someone tries to implement them again later (given Windows paths can be quite a bit more complicated than UNIX paths), and that might also result in scattered implementations across the codebase (manual fiddling with `/` and such). Let me know what you think... if you think it's worth the risk then I can remove it.

Reply:
Let's remove them for now, hopefully absl will soon have a filesystem library, then we can switch (https://github.com/abseil/abseil-cpp/issues/47). One can hope :D

Reply:
Sounds good haha.